### PR TITLE
Add docs for setUserTimezone()

### PR DESCRIPTION
**Pull request by markstory**
Refs cakephp/cakephp#15477

### CODE REVIEW

File: en/core-libraries/internationalization-and-localization.rst [hunk 1]
```diff
@@ -607,6 +607,57 @@ DateTime types to parse localized formats::
...
+
+When handling data from users in different timezones you will need to convert
+the datetimes in request data into your application's timezone. You can use
+``setUserTimezone()`` from a controller or :doc:`/controllers/middleware` to
+make this process simpler::
+
+    // Set the user's timezone
+    TypeFactory::build('datetime')->setUserTimezone($user->timezone);
+
+Once set, all datetime data will be treated as coming *from* the user's
```
Comment (othercorey) [inline en/core-libraries/internationalization-and-localization.rst]: This is misleading. Only marshalled data will be treated this way.

File: en/core-libraries/internationalization-and-localization.rst [hunk 2]
```diff
@@ -607,6 +607,57 @@ DateTime types to parse localized formats::
...
+timezone. During marshalling operations, the ORM will automatically convert
```
Comment (othercorey) [inline en/core-libraries/internationalization-and-localization.rst]:
```suggestion
timezone. When marshalling request data, the ORM will automatically convert
```

File: en/core-libraries/internationalization-and-localization.rst [hunk 3]
```diff
@@ -607,6 +607,57 @@ DateTime types to parse localized formats::
...
+        public function process(
+            ServerRequestInterface $request,
+            RequestHandlerInterface $handler
+        ): ResponseInterface {
+            // Get the user from the request.
+            $user = $requets->getAttribute('identity');
+            if ($user) {
+                TypeFactory::build('datetime')
+                    ->useLocaleParser()
+                    ->setUserTimezone($user->timezone);
```
Comment (othercorey) [inline en/core-libraries/internationalization-and-localization.rst:651]: where does the `$user->timezone` property come from?

File: en/core-libraries/internationalization-and-localization.rst [hunk 4]
```diff
@@ -607,6 +607,57 @@ DateTime types to parse localized formats::
...
+make this process simpler::
+
+    // Set the user's timezone
+    TypeFactory::build('datetime')->setUserTimezone($user->timezone);
+
+Once set, all datetime data will be treated as coming *from* the user's
+timezone. During marshalling operations, the ORM will automatically convert
+datetime values from the user's timezone into your application's timezone. This
+ensures that your application is always working in the timezone defined in
+``App.timezone``.
```
Comment (ADmad) [inline en/core-libraries/internationalization-and-localization.rst]:
```suggestion
``App.defaultTimezone``.
```

File: en/core-libraries/internationalization-and-localization.rst [hunk 5]
```diff
@@ -607,6 +607,57 @@ DateTime types to parse localized formats::
...
+    use Psr\Http\Server\RequestHandlerInterface;
+
+    class DatetimeMiddleare implements MiddlewareInterface
+    {
+        public function process(
+            ServerRequestInterface $request,
+            RequestHandlerInterface $handler
+        ): ResponseInterface {
+            // Get the user from the request.
+            $user = $requets->getAttribute('identity');
```
Comment (ADmad) [inline en/core-libraries/internationalization-and-localization.rst]:
```suggestion
            $user = $request->getAttribute('identity');
```

File: en/core-libraries/internationalization-and-localization.rst (hunk 3 above)
Comment (ADmad) [reply en/core-libraries/internationalization-and-localization.rst:651]: This is based on the assumption that users table has a `timezone` column.

**Pull request merged**